x-pack/plugins/security_solution/public/common/containers/sourcerer/index.tsx

```diff
@@ -84,8 +84,15 @@ export const useInitSourcerer = (
   );
   const { indexFieldsSearch } = useDataView();
 
+  const searchedIds = useRef<string[]>([]);
   useEffect(
-    () => activeDataViewIds.forEach((id) => id != null && id.length > 0 && indexFieldsSearch(id)),
+    () =>
+      activeDataViewIds.forEach((id) => {
+        if (id != null && id.length > 0 && !searchedIds.current.includes(id)) {
```

Comment:
whoops, we were calling this every time the activeDataViewIds changed even if we already had the data. now keeping track of what we've called already in `searchedIds`